pkgs/development/python-modules/netcdf4/default.nix

```
@@ -1,50 +1,72 @@
-{ stdenv, lib, buildPythonPackage, fetchPypi, isPyPy, pytest
+{ stdenv, lib, buildPythonPackage, fetchPypi, isPyPy
 , numpy, zlib, netcdf, hdf5, curl, libjpeg, cython, cftime
-}:
-buildPythonPackage rec {
-  pname = "netCDF4";
-  version = "1.5.3";
+,mpi4py ? null, openssh ? null}:
+assert (mpi4py != null) -> 
+  openssh != null
+  && hdf5.mpi == mpi4py.mpi 
+  && netcdf.mpi == mpi4py.mpi;
 
-  disabled = isPyPy;
+let 
+  mpiSupport = (mpi4py != null);
+  mpi= if mpiSupport then mpi4py.mpi else null;
+in
+  buildPythonPackage rec {
+    pname = "netCDF4";
+    version = "1.5.3";
+  
+    disabled = isPyPy;
+    
+    src = fetchPypi {
+      inherit pname version;
+      sha256 = "2a3ca855848f4bbf07fac366da77a681fcead18c0a8813d91d46302f562dc3be";
+    };
+  
+    checkInputs = [ netcdf numpy ] ++(if mpiSupport then [ mpi4py openssh mpi] else []);
+  
+    buildInputs = [
+      cython 
+      mpi 
+    ];
```

cython needs to be executed, should go in nativebuildinputs
```suggestion
    nativeBuildInputs = [
      cython
    ];
    
    buildInputs = [
      mpi 
    ];
```